src/activate.ts

```diff
@@ -27,9 +28,23 @@ async function Activate(): Promise<void> {
                 const servicesConfig = core.getInput('services-config', { required: true });
                 await licenseClient.ActivateLicenseWithConfig(servicesConfig);
             } else {
-                const username = core.getInput('username', { required: true }).trim();
-                const password = core.getInput('password', { required: true }).trim();
-                const serial = core.getInput('serial', { required: license.toLowerCase().startsWith('pro') });
+                const pro = license.toLowerCase().startsWith('pro');
+                let username = core.getInput('username', { required: pro }).trim();
+                let password = core.getInput('password', { required: pro }).trim();
+
+                if (!username && !password) {
+                    const encodedUsername = env['UNITY_USERNAME'];
```

Comment:
Some people can set this env variable by passing it in directly. We need to test if it is base64 encoded somehow.

Reply:
if they pass in in as inputs we'll treat it as a regular string. It's only if the inputs are missing that we treat it as base64. We could change the env var maybe UNITY_USERNAME_BASE64

Reply:
`UNITY_USERNAME_BASE64` I like this better

Reply:
I think it should be okay to pass your own ENV variables like UNITY_USERNAME or UNITY_PASSWORD. It would typically look something like this:

```yaml
      - uses: buildalon/activate-unity-license@v1
        with:
          license: 'Personal'
          UNITY_USERNAME: '${{ secrets.UNITY_USERNAME }}'
          UNITY_PASSWORD: '${{ secrets.UNITY_PASSWORD }}'
```

It is effectively the same